R/tar-stars.R

```diff
@@ -0,0 +1,136 @@
+                      storage = targets::tar_option_get("storage"),
+                      retrieval = targets::tar_option_get("retrieval"),
+                      cue = targets::tar_option_get("cue")) {
+
+    name <- targets::tar_deparse_language(substitute(name))
+
+    envir <- targets::tar_option_get("envir")
+
+    command <- targets::tar_tidy_eval(
+        expr = as.expression(substitute(command)),
+        envir = envir,
+        tidy_eval = tidy_eval
+    )
+
+    pattern <- targets::tar_tidy_eval(
+        expr = as.expression(substitute(pattern)),
+        envir = envir,
+        tidy_eval = tidy_eval
+    )
+
+    # if not specified by user, pull the corresponding geotargets option
+    driver <- driver %||% geotargets_option_get("gdal.raster.driver")
+    options <- options %||% geotargets_option_get("gdal.raster.creation_options")
+
+    targets::tar_target_raw(
+        name = name,
+        command = command,
+        pattern = pattern,
+        packages = packages,
+        library = library,
+        format = create_format_stars(driver = driver, options = options, ...),
+        repository = repository,
+        iteration = iteration,
+        error = error,
+        memory = memory,
+        garbage_collection = garbage_collection,
+        deployment = deployment,
+        priority = priority,
+        resources = resources,
+        storage = storage,
+        retrieval = retrieval,
+        cue = cue
+    )
+}
+#' @param driver character. File format expressed as GDAL driver names passed to [stars::write_stars()]. See [sf::st_drivers()].
+#' @param options character. GDAL driver specific datasource creation options passed to [stars::write_stars()]
+#' @param ... Additional arguments not yet used
+#' @noRd
+create_format_stars <- function(driver, options, ...) {
+
+    if (!requireNamespace("sf")) {
+        stop("package 'sf' is required", call. = FALSE)
+    }
+
+    if (!requireNamespace("stars")) {
+        stop("package 'stars' is required", call. = FALSE)
+    }
+
+    # get list of drivers available for writing depending on what the user's GDAL supports
+    drv <- sf::st_drivers(what = "raster")
+    drv <- drv[drv$write, ]
+
+    driver <- driver %||% geotargets_option_get("gdal.raster.driver")
+    driver <- rlang::arg_match0(driver, drv$name)
+
+    options <- options %||% geotargets_option_get("gdal.raster.creation_options")
+
+    # NOTE: Option getting functions are set in the .write_terra_raster function template
+    #       to resolve issue with body<- not working in some evaluation contexts ({covr}).
+    # TODO: It should be fine to have driver and options as NULL
+    .write_stars <- function(object, path) {
+        stars::write_stars(
+            object,
+            path,
+            driver = geotargets::geotargets_option_get("gdal.raster.driver"),
+            overwrite = TRUE,
+            options = geotargets::geotargets_option_get("gdal.raster.creation_options")
```

Comment:
Can't `stars` objects contain vectors?  How does this work with vector data cubes?

Reply:
Yes, good point, `stars` objects can contain vector geometry and an arbitrary number of dimensions. I said "new" issue but we can discuss this approach more in #14, I think